### PR TITLE
Separate Github runners for compute intensive tests.

### LINKED ISSUE #294: Creating separate github runners for compute intensive tests.
## Description
Since the test suites are intensive, we could create a separate runners for the compute intensive tests.

## Type of Test
- [ ] **Unit test** (e.g. checking a loop, method, or function is working as intended)

## Expected Behavior
A good improvement in CI time.

## Additional Context
Initially we could create a separate runner for [`mpc_tensor_test.py`](https://github.com/OpenMined/SyMPC/blob/main/tests/sympc/tensor/mpc_tensor_test.py), as they consume lot of time.The would deselecting `--deselect=sympc/tensor/mpc_tensor_test.py` in main workflow and create a separate workflow for mpc tensor tests by copying exact workflow file which would run mpc tensor tests.

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Created separate runners for compute intensive tests , resolves #294

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
CI Tests

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -82,9 +82,144 @@ jobs:
...
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-py${{ matrix.python-version }}-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-py${{ matrix.python-version }}-
+      - name: Install CPU Torch
+        if: runner.os != 'macos'
+        run: |
+          pip install torch==1.8.1+cpu torchvision==0.9.1+cpu torchcsprng==0.2.1+cpu -f https://download.pytorch.org/whl/torch_stable.html
```
Comment (gmuraru) [inline .github/workflows/tests.yml:124]: Q: Shouldn't we remove this? (we already have in the requirements files the requirements, right?)

Comment (rasswanth-s) [reply .github/workflows/tests.yml:124]: This change was by @madhavajay , pip packages contain only the gpu version, which are of large size, this downloads only the cpu version, shortening CI time.

Comment (gmuraru) [inline .github/workflows/tests.yml:124]: Q: Shouldn't we remove this? (we already have in the requirements files the requirements, right?)

**Review by gmuraru — APPROVED** [review]
LGTM!

**Pull request merged**